app/policies/api/tag_policy.rb

```diff
@@ -0,0 +1,10 @@
+module Api
+  # Policies for Api::Tag Controller
+  class TagPolicy < MainApiPolicy
+    alias_rule :create?, :update?, :destroy, to: :index
+    def index?
+      # has at least one instructor role
+      real_user.is_a?(EndUser) && real_user.roles.pluck(:type).include?('Instructor')
```

Comment:
This is too broad; you need to check that the current user's role _for the current course_ is `Instructor`. You should check how other API policies do this with the `role` variable